src/main/java/net/minecraftforge/fml/common/registry/VillagerRegistry.java

```diff
@@ -307,12 +340,30 @@ public boolean equals(Object o)
     public static void setRandomProfession(EntityVillager entity, Random rand)
     {
         Set<String> entries = INSTANCE.professions.getKeys();
-        int prof = rand.nextInt(entries.size());
-        //TODO: Grab id range from internal registry
-        entity.setProfession(rand.nextInt(5));
+        
+        // ===== Changes by dvatreknerd314 =====
+        // The goal here is to set the villager profession to a number between 0 and 5, but
+        // leaving non-vanilla options equally likely.
+        //
+        // If the profession is non-vanilla, the specific number selected is the (n-5)th key
+        // in the professions registry.
+        //
+        // at least, that's the goal
+        int prof = rand.nextInt(entries.size() + 5);
```

Comment:
Vanilla's entries should be treated exactly like normal entries.
The issue I was having was trying to design something that would provide predictable values for this.
Because professions are used in world generation. We need to make sure that things keep the same vanilla generation. There are a few possible ways to do this, however a lot of them are a PITA to do.